src/mqt/predictor/ml/predictor.py

```diff
@@ -339,40 +353,51 @@ def get_prepared_training_data(self) -> ml.helper.TrainingData:
         with resources.as_file(ml.helper.get_path_training_data() / "training_data_aggregated") as path:
             data = np.asarray(training_data, dtype=object)
             np.save(str(path / ("training_data_" + self.figure_of_merit + ".npy")), data)
-            data = np.asarray(names_list, dtype=str)
-            np.save(str(path / ("names_list_" + self.figure_of_merit + ".npy")), data)
-            data = np.asarray(scores_list, dtype=object)
-            np.save(str(path / ("scores_list_" + self.figure_of_merit + ".npy")), data)
+            if names_list:
+                data = np.asarray(names_list, dtype=str)
+                np.save(str(path / ("names_list_" + self.figure_of_merit + ".npy")), data)
+            if scores_list:
+                data = np.asarray(scores_list, dtype=object)
+                np.save(str(path / ("scores_list_" + self.figure_of_merit + ".npy")), data)
 
     def load_training_data(self) -> tuple[list[NDArray[np.float64]], list[str], list[NDArray[np.float64]]]:
         """Loads and returns the training data from the training data folder."""
+        training_data, names_list, scores_list = [], [], []
         with resources.as_file(ml.helper.get_path_training_data() / "training_data_aggregated") as path:
-            if (
-                path.joinpath("training_data_" + self.figure_of_merit + ".npy").is_file()
-                and path.joinpath("names_list_" + self.figure_of_merit + ".npy").is_file()
-                and path.joinpath("scores_list_" + self.figure_of_merit + ".npy").is_file()
-            ):
-                training_data = np.load(path / ("training_data_" + self.figure_of_merit + ".npy"), allow_pickle=True)
-                names_list = list(np.load(path / ("names_list_" + self.figure_of_merit + ".npy"), allow_pickle=True))
-                scores_list = list(np.load(path / ("scores_list_" + self.figure_of_merit + ".npy"), allow_pickle=True))
+            training_data_path = path / f"training_data_{self.figure_of_merit}.npy"
+            names_list_path = path / f"names_list_{self.figure_of_merit}.npy"
+            scores_list_path = path / f"scores_list_{self.figure_of_merit}.npy"
+
+            if training_data_path.is_file():
+                training_data = np.load(training_data_path, allow_pickle=True)
             else:
                 error_msg = "Training data not found. Please run the training script first as described in the documentation that can be found at https://mqt.readthedocs.io/projects/predictor/en/latest/Usage.html."
                 raise FileNotFoundError(error_msg)
 
+            # Names and scores are not mandatory: only used for evaluation
+            if names_list_path.is_file():
+                names_list = list(np.load(names_list_path, allow_pickle=True))
+
+            if scores_list_path.is_file():
+                scores_list = list(np.load(scores_list_path, allow_pickle=True))
+
             return training_data, names_list, scores_list
 
 
```

Comment:
Does the `predict_device_for_figure_of_merit` automatically work for the new figure of merit as well, even if it now is a regressor instead of a classifier?